src/modules/memory_usage.rs

```diff
@@ -1,90 +1,86 @@
-use ansi_term::Color;
-
-use super::{Context, Module};
 use byte_unit::{Byte, ByteUnit};
 use sysinfo::{RefreshKind, SystemExt};
 
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::memory_usage::MemoryConfig;
+
+fn format_kib(n_kib: u64) -> String {
+    let byte = Byte::from_unit(n_kib as f64, ByteUnit::KiB).unwrap_or_else(|_| Byte::from_bytes(0));
+    let mut display_bytes = byte.get_appropriate_unit(true).format(0);
+    display_bytes.retain(|c| c != ' ');
+    display_bytes
+}
+
 /// Creates a module with system memory usage information
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
-    const DEFAULT_THRESHOLD: i64 = 75;
-    const DEFAULT_SHOW_PERCENTAGE: bool = false;
-    const RAM_CHAR: &str = "🐏 ";
-
     let mut module = context.new_module("memory_usage");
+    let config = MemoryConfig::try_load(module.config);
+
+    // TODO: Update when v1.0 printing refactor is implemented to only
+    // print escapes in a prompt context.
+    let shell = std::env::var("STARSHIP_SHELL").unwrap_or_default();
+    let percent_sign = match shell.as_str() {
+        "zsh" => "%%", // % is an escape in zsh, see PROMPT in `man zshmisc`
+        "powershell" => "`%",
+        _ => "%",
+    };
 
-    if module.config_value_bool("disabled").unwrap_or(true) {
+    if config.disabled {
```

Comment:
This is automatically provided for all modules. No need to add the logic. 👍

Reply:
I think this might actually still be needed. https://github.com/starship/starship/pull/515#discussion_r333781415

When I was working on #561 I removed this line and started seeing the module appearing even though it was supposed to be disabled by default.

Reply:
No, disabling is currently not provided for modules unless a disable option is specified in the config. This is required as `memory` is disabled by default.